eth/downloader/queue.go

```diff
@@ -424,7 +424,7 @@ func (q *queue) ReserveHeaders(p *peerConnection, count int) *fetchRequest {
 	}
 	// Retrieve a batch of hashes, skipping previously failed ones
 	send, skip := uint64(0), []uint64{}
-	for send == 0 && !q.headerTaskQueue.Empty() {
+	for !q.headerTaskQueue.Empty() {
```

Comment:
If the "before" vs "after" of this change is "semantically and practically identical" - and it's only a matter of which syntax is nicest, then I prefer to not change it.